engine/runtime-with-polyglot/src/test/java/org/enso/interpreter/test/ForeignMethodInvokeTest.java

```diff
@@ -44,4 +47,66 @@ public void testForeignFunctionParseFailure() {
           e.getMessage().matches("Cannot parse foreign python method. Only available languages are .+"));
     }
   }
+
+  @Test
+  public void testInteropWithJavaScript() throws Exception {
+    var source = """
+        from Standard.Base import all
+
+        foreign js js_array t = \"\"\"
+            return [1, 2, t]
+
+        third t = js_array t
+        """;
+
+    var module = ctx.eval("enso", source);
+    var third = module.invokeMember("eval_expression", "third");
+    var res = third.execute(13);
+    assertTrue("It is an array", res.hasArrayElements());
+    assertEquals(3, res.getArraySize());
+    assertEquals(1, res.getArrayElement(0).asInt());
+    assertEquals(2, res.getArrayElement(1).asInt());
+    assertEquals(13, res.getArrayElement(2).asInt());
+
+    var res2 = Executors.newSingleThreadExecutor().submit(() -> {
+      return third.execute(12);
+    }).get();
+
+    assertTrue("It is an array2", res2.hasArrayElements());
+    assertEquals(12, res2.getArrayElement(2).asInt());
+  }
+
+  @Ignore
```

Comment:
I've reported https://github.com/oracle/graal/issues/8033 to describe why this test has to be `@Ignore`d right now.

Reply:
With `onDeniedThreadAccess` as proposed by https://github.com/oracle/graal/pull/8266 we have a way to [act like an Ethernet on collision](https://github.com/enso-org/enso/pull/8885/commits/160f2b81617f459a1a2bf198f940c78e7e7db9ed#diff-a84ceb8a331a33ef55d986aac04b55c48c52f2daddf6bea30911823ab24ff3a7R24) and successfully orchestrate the two threads accessing the same context to run into completion.

More info on [Ethernet collision detection and behavior](https://en.wikipedia.org/wiki/Carrier-sense_multiple_access_with_collision_detection) - most important _"Calculate and wait the random backoff"_.